### ISSUE TITLE
Update AWS Profiles to use t2.xlarge for masters

### ISSUE THREAD

**Issue by kris-nova** [kris-nova/kubicorn#262]
From master after deploying a cluster on `AWS`

```
kris-nova:kubicorn kris$ k get po --all-namespaces
NAMESPACE     NAME                                        READY     STATUS              RESTARTS   AGE
default       nginx-4217019353-mx0vp                      0/1       ContainerCreating   0          1h
kube-system   calico-etcd-4j77w                           1/1       Running             0          1h
kube-system   calico-node-1mfrm                           2/2       Running             0          1h
kube-system   calico-node-mp2g7                           1/2       CrashLoopBackOff    23         1h
kube-system   calico-policy-controller-1727037546-56gnf   1/1       Running             14         1h
kube-system   etcd-ip-10-0-0-94                           1/1       Running             0          1h
kube-system   kube-apiserver-ip-10-0-0-94                 1/1       Running             0          1h
kube-system   kube-controller-manager-ip-10-0-0-94        1/1       Running             0          1h
kube-system   kube-dns-2425271678-kn460                   0/3       ContainerCreating   0          1h
kube-system   kube-proxy-2ph5n                            1/1       Running             0          1h
kube-system   kube-proxy-cd4qg                            1/1       Running             0          1h
kube-system   kube-scheduler-ip-10-0-0-94                 1/1       Running             0          1h
```

So nothing is scheduling..

We need to fix this and solve integration testing sooner than later so this stops happening - we can't have master broken like this 😞

**Comment by bgeesaman** [comment]
It appears that calico-etcd is listening on the master's nodeIP (in my case 10.0.0.63), the calico config map is saying calico-etcd is on 10.96.x.x (the podIP), and the instance of calico-node on the worker cannot therefore connect to the endpoint http://10.96.x.x:6666.  I modified the calico-configmap to make the ETCD_ENDPOINT http://10.0.0.63:6666, and then killed the calico-node pod on the worker.  I then added tcp 6666 to the security group on master to allow the client to connect, and the calico-node pod stopped crashlooping and my test pods were able to then start on the worker.

I'm no kubeadm expert, but there appears to be a race condition of sorts that could be exacerbated by the fact that you're using t2.mediums which can get bogged down on cpu credits during the apt-get installs in your bootstrap scripts.

**Comment by kris-nova** [comment]
@bgeesaman tested and confirmed - intermittent on `t2.medium`s... I recently bumped our DO base install because load average was well over 12 during bootstrap..

Can we please bump our profiles to run `t2.xlarge` for masters?  I know it seems like a lot - but it's the first CPU bump in the tier

Changing issue title to reflect this action item, thanks for your discovery @bgeesaman :heart_eyes:

**Comment by ganlaksh** [comment]
I dont think there is an issue with the calico kube specification. I just had to open port 6666 on the security group of the master for the calico node to come up. 

calico-etcd has hostNetwork=true and hence i believe its gets the pod ip as the master node's ip. 
The calico-node config is trying to use the service or cluster ip of etcd which is 10.96.232.136

kube-proxy has this iptable rule to forward traffic to this ip to the master node ip and port 6666 

-A KUBE-SEP-44KVHP53FKAUSJ6N -p tcp -m comment --comment "kube-system/calico-etcd:" -m tcp -j DNAT --to-destination 10.0.0.183:6666

So , From kubicorn perspective , we just need to open tcp on port 6666 for the calico node to come up. 

For the application traffic , we still have unblock the udp traffic on ports 8285(flannel with udp backend) or 8472(flannel with vxlan). I am not sure what calico uses by default though(Should be vxlan). I am not sure about the ports though.

**Comment by bgeesaman** [comment]
I took a look at your calico.yml and it appears to spin etcd without the protection of TLS certificates.  I'd caution the exposure of calico-etcd on the master to pods in worker nodes unless you plan on implementing a networkpolicy to block that traffic from non-kube-system pods.  Otherwise, a compromised pod means it's easy to pull down a calicoctl binary and read/modify/disable those policies that might be protecting other critical pods, like the dashboard.

**Comment by ganlaksh** [comment]
@kris-nova I wil take this one..

**Comment by kris-nova** [comment]
@ganlaksh don't go too crazy just yet - I am doing a fairly major refactor in #299 and that might impact this PR

**Comment by kris-nova** [comment]
You are of course always welcome to open the PR, and I can cherry pick/rebase as needed - but just giving you a heads up

**Comment by matyix** [comment]
A workaround is to add this to your cluster description: 

```
{
	IngressFromPort: "0",
	IngressToPort:   "65535",
	IngressSource:   "10.0.100.0/24",
        IngressProtocol: "-1",
},
``` 

which allows all `TCP UDP ICMP` traffic from the `node` subnet. @kris-nova what was the reason behind using two different subnets for `master` and `node`? In Kops it's in the same subnet - not suggesting we need to change, just wondering if there is a specific reason.

**Comment by itsdimetrio** [comment]
@bgeesaman 
There is another thing from [calico documentation](https://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/hosted/kubeadm/):
`--pod-network-cidr` must be specified on kubeadm init command

**Comment by xmudrii** [comment]
Fixed with #332
